pkg/collector/python/check.go

```diff
@@ -51,6 +53,19 @@ type PythonCheck struct {
 	instanceConfig string
 }
 
+// It is an identical structure to the diagnosis.Diagnosis. It is possible
+// that in future diagnosis.Diagnosis will have its own JSON mapping but
+// at this stage this single usa case is insufficient to change base struct type
```

Comment:
This comment is slightly misleading, because we can't deserialize the base struct directly even if we wanted since it contains an `error` type (which JSON will refuse to deserialize).

But we can avoid having to duplicate the struct still: 1) as long as struct field names match JSON, annotations are not necessary, and 2) we can embed diagnosis into this struct and add only override the error field. The rest will get populated in the right places automatically: https://go.dev/play/p/RV6JfmKH-yo

Reply:
I was not aware of these nuances (thank you for providing playground code). I will try to change implementation to be more inline with this suggestion. Thank you.